lib/storage/src/content_manager/consensus_state.rs

```diff
@@ -309,6 +309,7 @@ impl ConsensusState {
             state.hard_state.term = cmp::max(state.hard_state.term, meta.term);
             state.hard_state.commit = meta.index
         })?;
+        self.persistent.write().snapshot_applied()?;
```

Comment:
Here we are acquiring the persistent write lock twice in a row, maybe we could reused the guard instead?